### ISSUE TITLE
accumulate_grad_batches ignores last batches in epoch if number of steps is not divisible by accumulate_grad_batches?

### ISSUE THREAD

**Issue by Vozf** [PyTorchLightning/pytorch-lightning#4653]
Suppose I have accumulate_grad_batches=256 and number of steps in my epoch is 260. Loss is updated only on step number 256 every epoch. I suppose it means that the last 4 batches grads are ignored. Is that correct?

**Comment by ydcjeff** [comment]
I suppose we do not ignore for last batches. Can you share a minimal example if it's not working?

**Comment by Vozf** [comment]
So what is done with last 6 batches? Is it aggreagated over 6 batches instead of asked 256?

**Comment by ydcjeff** [comment]
We call `.backward` and `optimizer.step` `optimizer.zero_grad()` for the last 4 batches.

**Comment by Vozf** [comment]
So first you accumulate 256 batches and call backward and then you accumulate 4 batches and call backward, correct?

**Comment by ydcjeff** [comment]
Yep we accmulate 256 if possible and accumulate the rest of batches if it's not divisible by 256

**Comment by Vozf** [comment]
Ok, thanks for clarification
